ui-v2/src/components/ui/combobox/combobox.tsx

```diff
@@ -0,0 +1,155 @@
+import { Button } from "@/components/ui/button";
+import {
+	Command,
+	CommandEmpty,
+	CommandGroup,
+	CommandInput,
+	CommandItem,
+	CommandList,
+} from "@/components/ui/command";
+import { Icon } from "@/components/ui/icons";
+import {
+	Popover,
+	PopoverContent,
+	PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { createContext, use, useState } from "react";
+import { FormControl } from "../form";
+
+const ComboboxContext = createContext<{
+	open: boolean;
+	setOpen: (open: boolean) => void;
+} | null>(null);
+
+const Combobox = ({ children }: { children: React.ReactNode }) => {
+	const [open, setOpen] = useState(false);
+	return (
+		<ComboboxContext.Provider value={{ open, setOpen }}>
+			<Popover open={open} onOpenChange={setOpen}>
+				{children}
+			</Popover>
+		</ComboboxContext.Provider>
+	);
+};
+
+const ComboboxTrigger = ({
+	selected = false,
+	withForm = false,
+	children,
+}: { selected?: boolean; withForm?: boolean; children: React.ReactNode }) => {
+	const comboboxCtx = use(ComboboxContext);
+	if (!comboboxCtx) {
+		throw new Error("'ComboboxContext' not provided");
+	}
```

Comment:
I think this error could be a little more helpful.
```suggestion
	if (!comboboxCtx) {
		throw new Error("'ComboboxTrigger' must be a child of `Combobox`");
	}
```

Reply:
Updated in latest push